### PR TITLE
(dev/core#624) Price Set Custom search: add Participant status as filter

### PR DESCRIPTION
**Pull request by sunilpawar**
Overview
----------------------------------------
Add Participant status as filter to custom search

Before
----------------------------------------
Status field was absent in filter criteria as well in result set.

After
----------------------------------------
Status Field is present (mult-select) as field and column in result set.

price_set_status_filter

https://lab.civicrm.org/dev/core/issues/624

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@colemanw where are you at on this - do we actually accept new features to the build in custom searches? Ideally instead it would be removed from core & go in an extension where people can add & maintain - where are you at with this?

I would note there is no test & while it perhaps shouldn't be reviewed without one - if the review is going to be to say 'do it in an extension' we should do that before asking for a test....

File: CRM/Contact/Form/Search/Custom/PriceSet.php [hunk 1]
```diff
@@ -316,6 +354,7 @@ public function from() {
     $from = "
 FROM       civicrm_contact contact_a
 INNER JOIN {$this->_tableName} tempTable ON ( tempTable.contact_id = contact_a.id ) {$this->_aclFrom}
+INNER JOIN civicrm_participant_status_type participant_status ON (tempTable.status_id = participant_status.id)
```
Comment (colemanw) [inline CRM/Contact/Form/Search/Custom/PriceSet.php]: Doing this as a join may hurt performance. In other searches we look up the options with PHP when rendering results.

File: CRM/Contact/Form/Search/Custom/PriceSet.php [hunk 2]
```diff
@@ -212,6 +235,16 @@ public function buildForm(&$form) {
       TRUE
     );
 
+    $form->addSelect('status_id',
+      array(
+        'entity' => 'participant',
+        'label' => ts('Participant Status'),
+        'multiple' => 'multiple',
```
Comment (colemanw) [inline CRM/Contact/Form/Search/Custom/PriceSet.php]: Typically we say `'multiple' => TRUE`

**Comment by colemanw** [comment]
I think the custom search ought to go in an extension but for now it's in core so this PR is appropriate. Merging this PR doesn't stop us from moving the search to an extension at some point.

**Review by colemanw — CHANGES_REQUESTED** [review]
I'd be inclined to merge this if it can be implemented without an extra join.

**Comment by eileenmcnaughton** [comment]
@colemanw I think we should get a test too - if something breaks we still have to deal with the fallout so I don't think we should drop the test requirement in the under-maintained & flakey sections of code - esp not for an additional feature rather than a bug fix

**Comment by eileenmcnaughton** [comment]
Closing to track through gitlab as this is stale, without a test & review is not addressed - treating as inactive for now

**Pull request closed without merging**